cdap-watchdog/src/main/java/co/cask/cdap/logging/read/KafkaLogReader.java

```diff
@@ -44,6 +44,8 @@
 public class KafkaLogReader implements LogReader {
   private static final Logger LOG = LoggerFactory.getLogger(KafkaLogReader.class);
   private static final int KAFKA_FETCH_TIMEOUT_MS = 30000;
+  // Maximum events to read from Kafka for any call
+  private static final int MAX_READ_EVENTS_KAFKA = 10000;
```

Comment:
as we discussed, if this limit is being hit, changing the number of log partitions would fix the underlying issue, so we will not have a configuration setting for this.